### PR TITLE
feat: graphql-kotlin-dataloader

### PR DESCRIPTION
**Pull request by samuelAndalon**
### :pencil: Description
Extracting DataLoader into his own module - This new module will decorate the DataLoaderRegistry to have access to the each DataLoader futures

### CODE REVIEW

File: executions/graphql-kotlin-dataloader/README.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
+# GraphQL Kotlin Data Loader
+[![Maven Central](https://img.shields.io/maven-central/v/com.expediagroup/graphql-kotlin-dataloader.svg?label=Maven%20Central)](https://search.maven.org/search?q=g:%22com.expediagroup%22%20AND%20a:%22graphql-kotlin-dataloader%22)
+[![Javadocs](https://img.shields.io/maven-central/v/com.expediagroup/graphql-kotlin-dataloader.svg?label=javadoc&colorB=brightgreen)](https://www.javadoc.io/doc/com.expediagroup/graphql-kotlin-dataloader)
+
+Data Loaders are a popular caching pattern from the [JavaScript GraphQL implementation](https://github.com/graphql/dataloader).
+
+Since `graphql-java` provides [support for this pattern](https://www.graphql-java.com/documentation/v16/batching/)
```
Comment (dariuszkuc) [inline executions/graphql-kotlin-dataloader/README.md:7]: should we link to the v18 instead? https://www.graphql-java.com/documentation/batching

File: executions/graphql-kotlin-dataloader/README.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+using the [DataLoader](https://github.com/graphql-java/java-dataloader/blob/master/src/main/java/org/dataloader/DataLoader.java)
+and [DataLoaderRegistry](https://github.com/graphql-java/java-dataloader/blob/master/src/main/java/org/dataloader/DataLoaderRegistry.java).
+
+`graphql-kotlin` provides support for `DataLoaders` with this `graphql-kotlin-dataloader` module through common interfaces
```
Comment (dariuszkuc) [inline executions/graphql-kotlin-dataloader/README.md:11]:
```suggestion
`graphql-kotlin` provides support for `DataLoaders` with this `graphql-kotlin-dataloader` module through common interfaces.
```

**Review by dariuszkuc — CHANGES_REQUESTED** [review]
Should we also update https://opensource.expediagroup.com/graphql-kotlin/docs/server/data-loaders with the info about importing this module?

**Comment by samuelAndalon** [comment]
> Should we also update https://opensource.expediagroup.com/graphql-kotlin/docs/server/data-loaders with the info about importing this module?

The server would be importing it by default, not sure if its worth to add that info in the documentation

**Comment by dariuszkuc** [comment]
> > Should we also update https://opensource.expediagroup.com/graphql-kotlin/docs/server/data-loaders with the info about importing this module?
> 
> The server would be importing it by default, not sure if its worth to add that info in the documentation
> 
Technically you could import it without using our server packages.

**Review by dariuszkuc — APPROVED** [review]

**Pull request merged**